tests/shared_test_functions.sh

```diff
@@ -108,16 +108,32 @@ has_pid() {
 	return 1
 }
 
-## wait_for_file (filename):
-# Waits until ${filename} exists.
-wait_for_file() {
-	filename=$1
-	while [ ! -e "${filename}" ]; do
+## wait_while(func):
+# Wait while ${func} returns 0.  If ${msleep} is defined, use that to wait
+# 100ms; otherwise, wait in 1 second increments.
+wait_while() {
+	_wait_while_ms=0
+
+	# Check for the ending condition
+	while "$@"; do
+		# Notify user (if desired)
 		if [ "${VERBOSE}" -ne 0 ]; then
-			echo "Waiting for ${filename}" 1>&2
+			printf "waited\t%ims\t%s\n"		\
+			    "${_wait_while_ms}" "$*" 1>&2
+		fi
+
+		# Wait using the appropriate binary
+		if [ -n "${msleep:-}" ];  then
```

Comment:
libcperciva does not have an `${msleep}` variable defined, and at I have no current plans to import our `msleep` binary to it.

However, `msleep` is currently used in kivaloo, and I'll be importing it to spiped, so it's worth including this in the generic `shared_test_functions.sh` file.